File: Components/Report/DrilldownState.vb
Namespace DNNStuff.SQLViewPro

    <Serializable()> _
    Public Class DrilldownState
        Public Property FromReportId() As Integer = -1
        Public Property FromReportColumn() As String = ""
        Public Property Parameters() As ArrayList
        Public Property ReportSet() As ReportSetInfo
        Public Property PortalId() As Integer
        Public Property ModuleId() As Integer
        Public Property TabId() As Integer
        Public Property UserId() As Integer

        Public Sub New(ByVal FromReportId As Integer, ByVal FromReportColumn As String, ByVal Parameters As ArrayList)
            _FromReportId = FromReportId
            _FromReportColumn = FromReportColumn
            _Parameters = Parameters
        End Sub
        Public Sub New(ByVal Parameters As ArrayList)
            _Parameters = Parameters
        End Sub
    End Class

End Namespace

File: Components/Report/DrilldownState.cs
using System;
using System.Collections;

namespace DNNStuff.SQLViewPro
{
	
	[Serializable()]public class DrilldownState
	{
	    public int FromReportId { get; set; } = -1;
	    public string FromReportColumn { get; set; } = "";
        public ArrayList Parameters { get; set; } = new ArrayList();
		public ReportSetInfo ReportSet {get; set;}
		public int PortalId {get; set;}
		public int ModuleId {get; set;}
		public int TabId {get; set;}
		public int UserId {get; set;}
		
		public DrilldownState(int fromReportId, string fromReportColumn, ArrayList parameters)
		{
			FromReportId = fromReportId;
			FromReportColumn = fromReportColumn;
			Parameters = parameters;
		}
		public DrilldownState(ArrayList parameters)
		{
			Parameters = parameters;
		}
	}
	
}

